### DISCUSSION TITLE
Including "Array" type for the configuration file Value

### DISCUSSION THREAD

**Discussion by MartaUch** [Azure-Player/azure.datafactory.tools#362] [Q&A] [answered]
Hello,
Currently I can see in the documentation that only three types of the Value can be accepted: string, number and nested json object.
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/0df75bd5-aa5e-41c1-b9e3-be311fed773d)
Can we please extend this list with Array type as well?

We want to update global parameter that originally suppose to be an array and when we define value this way in the json config file:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/bd15e4ba-22a9-460f-b79a-8b68101896f8)
Then the array is deployed, but with these single quotation marks what is not acceptable by ADF:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/d5723f2c-1f1b-483e-a826-9c96d0d4015d)
It requires this format:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/721c8282-9389-439e-9939-e77bf98e9c4a)
We also tried with "value": ["dev poc","test"] in the config file but then we are getting: 
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/96fab250-409c-475c-8345-e7e8c864a0c4)

- Is there any workaround for this to enable deployment of the arrays? 
- If not, is it possible to extend list of the acceptable Value types?

That is actually the last thing that stops me from using incremental deployment. I want to be sure that developers will be capable of defining and updating Array types for global parameters.
Best regards
Marta

**Comment by MartaUch** [comment]
Alright, Now I noticed that in the old issues I can find some discussions already opened. I apologies, I will take a look at them first

**Comment by MartaUch** [comment]
I checked some of the issues, but to be honest I still can't find correct approach. I switched to csv now and used the following approaches. This is my factory file:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/00563cc1-107b-4628-bea4-3c4e556ef2d4)
This is my csv config file:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/a5252add-39a3-4e61-8468-e392e347cfb4)
This is an error for the Array:
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/f850f401-1610-48a4-821b-d6b3e4cceeb0)
Same error occurs then value is defined as "[5,6,7]"
Thank you in advance for your help :)

**Comment by MartaUch** [comment]
Hello Kamil :) 
Is there any update on this issue?

**Comment by NowinskiK** [comment]
Not yet, I'm on holiday right now.

**Comment by MartaUch** [comment]
Alright, I apologise, wish you a nice holiday!

**Comment by NowinskiK** [comment, marked as answer]
I investigated the issue and found solution. In order to define an array correctly you must avoid using square brackets.
So, instead of this:
```
type,name,path,value
factory,BigFactorySample2,"$.properties.globalParameters.'GP-Array'.value","[2,4,""XYZ"",1000]"
```

**Do this - correct format for array (in PowerShell):**
```
type,name,path,value
factory,BigFactorySample2,"$.properties.globalParameters.'GP-Array'.value","2,4,""XYZ"",1000"
```

**Reply by MartaUch** [reply]
Hi Kamil,
Thank you for this, I will test it as soon as possible and get back to you :)

**Reply by MartaUch** [reply]
Hi Kamil,
I had a chance to test this approach today and it works as suppose to :)
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/d8d98f23-564f-48a7-add0-25032165c8b6)
![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/28823253/d1049608-61e3-42c7-9553-8126797fc7f2)

Thank you!
Marta

**Reply by Antranig7** [reply]
> I investigated the issue and found solution. In order to define an array correctly you must avoid using square brackets. So, instead of this:
> 
> ```
> type,name,path,value
> factory,BigFactorySample2,"$.properties.globalParameters.'GP-Array'.value","[2,4,""XYZ"",1000]"
> ```
> 
> **Do this - correct format for array (in PowerShell):**
> 
> ```
> type,name,path,value
> factory,BigFactorySample2,"$.properties.globalParameters.'GP-Array'.value","2,4,""XYZ"",1000"
> ```

Thank you for this, it really helps! It's working nicely when there are more than 1 value in the array, but not whenever there's only 1 value in the array. Example use case: this would be needed when VMs are being controlled via the adf, and in some environments you need only 1 VM (1 value in the array) and in other environments, you need several VMs. I tried adding a , after the single value like: """vm1""," but this fails with missing expression after comma error.
